src/hardwareLayer/io/SignalGenerator.cpp

```diff
@@ -100,8 +110,32 @@ Signal SignalGenerator::nextSignal() {
 	return signal;
 }
 
+bool SignalGenerator::dealWithClatter(Signal signal){
+	auto timeSinceLastInterrupt = std::chrono::duration_cast <std::chrono::milliseconds> (std::chrono::steady_clock::now() - time);
+	time = std::chrono::system_clock::now();
+	if  (signal.name == Signalname::BUTTON_E_STOP_PUSHED){
+		if (timeSinceLastInterrupt.count() <= E_STOP_CLUTTER_TIME){
+			return false;
+		}
+	}
+	else if (timeSinceLastInterrupt.count() <= SWITCH_BUTTON_CLUTTER_TIME){
```

Comment:
Das man die Buttons auf einer selben Zeit hat finde ich okay. Die Switch allerdings nicht unbedingt. Das sehe ich jetzt nicht so eng wie mit dem E_STOP, aber wenn jemand auf STOP drückt und deswegen ein SWITCH_OPEN Signal nicht nach oben gereicht wird werden wir in einen Fehlerfall kommen "Switch konnte nicht geoeffnet werden", obwohl sie offen ist.

Das ist definitiv ein Edge Case , der echt nicht häufig auftreten wird aber dafür umso ärgerlicher ist..